mpi-proxy-split/seq_num.cpp

```diff
@@ -44,15 +44,15 @@ int comm_resume_target_reached(unsigned int global_comm) {
 
 void seq_num_init() {
   ckpt_pending = false;
-  freepass = false;
   pthread_mutex_init(&seq_num_lock, NULL);
   sem_init(&user_thread_sem, 0, 0);
   sem_init(&ckpt_thread_sem, 0, 0);
+  sem_init(&freepass_sem, 0, 0);
 }
 
 void seq_num_reset(reset_type_t type) {
   ckpt_pending = false;
-  freepass = false;
+  while(sem_trywait(&freepass_sem) == 0);
```

Comment:
FORMATTING:  `while (sem_trywait ...)` with space after 'while'.

Reply:
We still need a space after `while` in `while (sem_trywait ...)` on the line above.